File: Backend/shipment/shipment/src/main/java/com/shipfast/shipment/entity/PricingConfig.java
package com.shipfast.shipment.entity;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Document(collection = "pricing_configs")
public class PricingConfig {

    @Id
    private String id;

    private double standardRatePerKg;
    private double expressMultiplier;
    private double sameDayMultiplier;
    private double distanceSurcharge;
    private double fuelSurchargePct;
    private double gstPct;
    private double codHandlingFee;
}

File: mern-backend/src/models/PricingConfig.js
import mongoose from 'mongoose';

const PricingConfigSchema = new mongoose.Schema(
  {
    _id: {
      type: String,
      default: 'DEFAULT'
    },
    standardRatePerKg: {
      type: Number,
      default: 80.0
    },
    expressMultiplier: {
      type: Number,
      default: 1.75
    },
    sameDayMultiplier: {
      type: Number,
      default: 2.0
    },
    distanceSurcharge: {
      type: Number,
      default: 40.0
    },
    fuelSurchargePct: {
      type: Number,
      default: 9.0
    },
    gstPct: {
      type: Number,
      default: 5.0
    },
    codHandlingFee: {
      type: Number,
      default: 50.0
    }
  },
  {
    timestamps: true,
    _id: false
  }
);

PricingConfigSchema.methods.toDto = function () {
  return {
    standardRatePerKg: this.standardRatePerKg,
    expressMultiplier: this.expressMultiplier,
    sameDayMultiplier: this.sameDayMultiplier,
    distanceSurcharge: this.distanceSurcharge,
    fuelSurchargePct: this.fuelSurchargePct,
    gstPct: this.gstPct,
    codHandlingFee: this.codHandlingFee
  };
};

export const PricingConfig = mongoose.models?.PricingConfig || mongoose.model('PricingConfig', PricingConfigSchema);
